ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -164,6 +166,8 @@ class GetBlobOperation extends GetOperation {
         shouldLookupMetadataCache() ? (BlobMetadata) blobMetadataCache.getObject(blobId.toString()) : null;
     firstChunk = (blobMetadata == null) ? new FirstGetChunk() : new CachedFirstChunk(blobMetadata);
     this.nonBlockingRouter = nonBlockingRouter;
+    decompressionService = new CompressionService(routerConfig.getCompressionConfig(),
```

Comment:
So for each GET, we create new instance of compression service ?
Can we do better - like having single shared instance of this service and enqueueing requests to it like how encryption/decryption work ?

Reply:
actually, nevermind. This is ok.

Reply:
Eventually, when we adopt dependency injection.  It should be a singleton.